src/components/projects/wbs-details/work-package-container/work-package-details/work-package-details.tsx

```diff
@@ -22,29 +19,75 @@ interface WorkPackageDetailsProps {
 const WorkPackageDetails: React.FC<WorkPackageDetailsProps> = ({ workPackage }) => {
   const detailsBody = (
     <Container fluid>
-      <Row>
-        <Col xs={12} md={6}>
-          <b>Work Package Name:</b> {workPackage.name} <br />
-          <b>WBS #:</b> {wbsPipe(workPackage.wbsNum)} <br />
-          <b>Project Lead:</b> {listPipe(workPackage.projectLead, fullNamePipe)} <br />
-          <b>Project Manager:</b> {fullNamePipe(workPackage.projectManager)} <br />
-          <b>Duration:</b> {weeksPipe(workPackage.duration)} <br />
-        </Col>
-        <Col xs={6} md={4}>
-          <b>Start Date:</b> {workPackage.startDate.toLocaleDateString()} <br />
-          <b>End Date:</b> {endDatePipe(workPackage.startDate, workPackage.duration)} <br />
-          <b>Progress:</b> {workPackage.progress}% <br />
-          <b>Expected Progress:</b> <br />
-          <b>Timeline Status:</b> <br />
-        </Col>
-      </Row>
+      <Form>
+        <Row>
+          <Col xs={12} md={6}>
+            <EditableDetail title="Work Package Name" value={workPackage.name} type="text" />
+            <EditableDetail title="WBS #" value={wbsPipe(workPackage.wbsNum)} type="text" />
+            <EditableDetail
+              title="Project Lead"
+              value={listPipe(workPackage.projectLead, fullNamePipe)}
+              type="text"
+            />
+            <EditableDetail
+              title="Project Manager"
+              value={fullNamePipe(workPackage.projectManager)}
+              type="text"
+            />
+            <EditableDetail
+              title="Duration"
+              value={`${workPackage.duration}`}
+              type="number"
+              suffix="weeks"
+            />
+          </Col>
+          <Col xs={6} md={4}>
+            <EditableDetail
+              title="Start Date"
+              value={workPackage.startDate.toLocaleDateString()}
+              type="date"
+            />
+            <EditableDetail
+              title="End Date"
+              value={endDatePipe(workPackage.startDate, workPackage.duration)}
+              type="date"
+              readOnly={true}
+            />
+            <EditableDetail
+              title="Progress"
+              value={`${workPackage.progress}`}
+              suffix="%"
+              type="number"
+            />
+          </Col>
+        </Row>
+      </Form>
     </Container>
   );
 
+  const editMode = useContext(EditModeContext);
+  const statuses = ['Inactive', 'Active', 'Complete'];
```

Comment:
enum already exists in `utils` package, so you should be able to just do `Object.values(...)`
https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/dca5d7b78d63c608c9cac301fdd9b27361f35cd4/src/utils/src/types/project-types.ts#L27